iidm/iidm-api/src/main/java/com/powsybl/iidm/network/util/SV.java

```diff
@@ -18,55 +21,45 @@
  * the state variables on the other side.
  *
  * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ * @author Luma Zamarreño <zamarrenolm at aia.es>
+ * @author José Antonio Marqués <marquesja at aia.es>
  */
 public class SV {
 
-    public static double getRho(TwoWindingsTransformer twt) {
-        return twt.getRatedU2() / twt.getRatedU1() * twt.getOptionalRatioTapChanger().map(rtc -> rtc.getCurrentStep().getRho()).orElse(1d);
-    }
-
-    public static double getR(TwoWindingsTransformer twt) {
-        return twt.getR() * twt.getOptionalRatioTapChanger().map(rtc -> 1 + rtc.getCurrentStep().getR() / 100).orElse(1d);
-    }
-
-    public static double getX(TwoWindingsTransformer twt) {
-        return twt.getX() * twt.getOptionalRatioTapChanger().map(rtc -> 1 + rtc.getCurrentStep().getX() / 100).orElse(1d);
-    }
+    //
+    // | Y11 Y12 ||V1|   |I1|
+    // |         ||  | = |  |
+    // | Y21 Y22 ||V2|   |I2|
+    //
+    // I1 = S1* / V1* and I2 = S2* / V2*
+    //
 
-    public static double getG(TwoWindingsTransformer twt) {
-        return twt.getG() * twt.getOptionalRatioTapChanger().map(rtc -> 1 + rtc.getCurrentStep().getG() / 100).orElse(1d);
+    public SV(double p, double q, double u, double a, Branch.Side side) {
+        this.p = p;
+        this.q = q;
+        this.u = u;
+        this.a = a;
+        this.side = side;
     }
 
-    public static double getB(TwoWindingsTransformer twt) {
-        return twt.getB() * twt.getOptionalRatioTapChanger().map(rtc -> 1 + rtc.getCurrentStep().getB() / 100).orElse(1d);
+    /**
+     * @deprecated Not used anymore. The end associated to the voltage and flow must be defined. Use
+     * {@link SV#SV(double, double, double, double, int)}
+     */
+    @Deprecated
+    public SV(double p, double q, double u, double a) {
+        throw new PowsyblException("Deprecated. Not used anymore");
```

Comment:
If you deprecate it, don't throw exception. Log a warning explaining it always uses side ONE and return SV(p, q, u, a, ONE)

Reply:
We prefer to throw an exception to force the user to review the code. In some cases the difference could be small and the best way to be sure that the code is ok is to review it.